qms-tests/sanity/tests/documents/ES-40.spec.ts

```diff
@@ -0,0 +1,54 @@
+import { test } from '@playwright/test'
+import { attachScreenshot, HomepageURI, PlatformSettingSecond, PlatformURI } from '../utils'
+import { allure } from 'allure-playwright'
+import { DocumentContentPage } from '../model/documents/document-content-page'
+import { faker } from '@faker-js/faker'
+
+test.use({
+  storageState: PlatformSettingSecond
+})
+
+test.describe('ISO 13485, 4.2.4 Control of documents ensure that documents of external origin are identified and their distribution controlled', () => {
+  test.beforeEach(async ({ page }) => {
+    await (await page.goto(`${PlatformURI}/${HomepageURI}`))?.finished()
+  })
+
+  test('TESTS-391. As a non space member, I cannot see nor edit any doc from that space', async ({ page }) => {
+    await allure.description(
+      'Requirement\nUser is not a part of space members and cannot see or edit any document from that space'
+    )
+    await allure.tms('TESTS-391', 'https://tracex.hc.engineering/workbench/platform/tracker/TESTS-391')
+    await test.step('2. check if non member can see space', async () => {
+      const folderName = faker.word.words(1)
+      const documentContentPage = new DocumentContentPage(page)
+      await documentContentPage.clickAddFolderButton()
+      await documentContentPage.fillDocumentSpaceFormManager(folderName)
+      await documentContentPage.changeDocumentSpaceMembers(folderName)
+      await documentContentPage.checkIfTheSpaceIsVisible(folderName, false)
+      await documentContentPage.clickDocumentsSpace()
+      await page.waitForTimeout(1000)
```

Comment:
Why do we need this timeout?

Reply:
unfortunately we need, I noticed sometimes the test is flaky without it

Reply:
we can increase the max timeout only, but it still sees this element right away so if we increase  the functionality of 

`const teamspaceOrArrow = await this.page.isVisible('.w-full > button:nth-child(2)')
`

it will just increase the max timeout to wait.

Reply:
I assumed that it was failing there w/o the explicit timeout. Where it was failing then?

Reply:
it just skipped the step, like it clicked before the page was rendered. Let me try something